src/Storages/NamedCollectionsHelpers.cpp

```diff
@@ -105,7 +105,7 @@ MutableNamedCollectionPtr tryGetNamedCollectionWithOverrides(
             if (allow_override_by_default)
                 continue;
             // if allow_override_by_default is false we don't allow extra arguments
-            throw Exception(ErrorCodes::BAD_ARGUMENTS, "Override not allowed");
+            throw Exception(ErrorCodes::BAD_ARGUMENTS, "Override not allowed because allow_override_by_default is false");
```

Comment:
Maybe sounds a bit better :)
```suggestion
            throw Exception(ErrorCodes::BAD_ARGUMENTS, "Override not allowed because setting allow_override_by_default is disabled");
```